src/OCC/valise/valise.cpp

```diff
@@ -255,4 +263,54 @@ void routeValiseCborEcho(OSCMessage &msg, int addressOffset)
     msg2.add(
         "d08355a20101055001010101010101010101010101010101a10458246d65726961646f632e6272616e64796275636b406275636b6c616e642e6578616d706c655820c4af85ac4a5134931993ec0a1863a6e8c66ef4c9ac16315ee6fecd9b2e1c79a1");
     sendOSCMessage(msg2);
-} 
\ No newline at end of file
+} 
+
+/**
+ * Store the address family inside the device's memory
+ *
+ * @param String(0) The address family, either "tb" for testbtc or "bc" for btc.
+ * @return Generated '0' or '1' string for failure or success. Sending over OSC as string
+ */
+void routeValiseAddrFamily(OSCMessage &msg, int addressOffset)
+{
+    Preferences valise;
+    if (msg.isString(0))
+    {
+        int length = msg.getDataLength(0);
+        char addrFamily[length];
+        msg.getString(0, addrFamily, length);
+
+        // Validate address family
+        if (strcmp(addrFamily, "tb") == 0 || strcmp(addrFamily, "bc") == 0) {
+            valise.begin("vault", false);
+            valise.putString("addr_family", addrFamily);
+            valise.end();
+
+            OSCMessage resp_msg("/valiseAddrFamilySet");
+            resp_msg.add("1"); // Success
+            sendOSCMessage(resp_msg);
+        } else {
+            // Invalid address family
+            OSCMessage resp_msg("/valiseAddrFamilyError");
+            resp_msg.add("0"); // Failure
+            sendOSCMessage(resp_msg);
+        }
+    }
+}
+
+/**
+ * Retrieve the address family from the device's memory
+ *
+ * @return The stored address family. Sending over OSC as string
+ */
+void routeValiseAddrFamilyGet(OSCMessage &msg, int addressOffset)
+{
+    Preferences valise;
+    valise.begin("vault", false);
+    String addrFamily = valise.getString("addr_family", "bc"); // Default to "bc" if not set
```

Comment:
Should there be a 'bc' parameter?
or jut a comment?

Reply:
this is for the differentiation of mainnet and tested for libwally core functions